documentation/asciidoc/topics/proc_changing_operator_password.adoc

```diff
@@ -0,0 +1,18 @@
+[id='changing-operator-password_{context}']
+= Changing the Operator Password
+You can change the password for the `operator` user if you do not want to use the automatically generated password.
+
+.Procedure
+
+* Update the `password` key in the `{example_crd_name}-generated-operator-secret` secret as follows:
++
+[source,bash,options="nowrap",subs=attributes+]
+----
+{oc} patch secret {example_crd_name}-generated-operator-secret -p='{"stringData":{"password": "supersecretoperatorpassword"}}'
+----
++
+[NOTE]
+====
+You should update only the `password` key in the `generated-operator-secret` secret.
+When you update the password, {ispn_operator} automatically refreshes other keys in that secret.
```

Comment:
@dmvolod @ryanemerson What happens at this point, is the whole cluster restarted?

Prometheus has side container that's watching for the changes in the resources provided by secrets/configmaps and when the change happens it triggers config reload on server so it doesn't restart the nodes.

Reply:
@Crumby this is related not only for Prometheus, Infinispan cluster need to be restarted when admin password has been changed

Reply:
Also I can't get Prometheus side container works without ServiceMonitor update. I tried to change Infinispan cluster password, but after that Prometheus metrics never been collected

Reply:
> I can't get Prometheus side container works without ServiceMonitor update

Right, that's not good, is the ServiceMonitor recreated then so the Prometheus catches up or is there anything needed on user side?

My initial idea was about the need to restart the server though and I used Prometheus CR as an example of implementation which might caused misunderstanding.

Infinispan server itself doesn't need to be restarted to catch on updated credentials. If we had side container catching up on the secret changes we could trigger config-generator reload which would update all user.properties file on fly while server would be still up and running. But this is way out of scope of this PR

Reply:
> Right, that's not good, is the ServiceMonitor recreated then so the Prometheus catches up or is there anything needed on user side?

Nothing need, I updated ServiceMonitor CR with the password hash annotation

Reply:
> > I can't get Prometheus side container works without ServiceMonitor update
> 
> Right, that's not good, is the ServiceMonitor recreated then so the Prometheus catches up or is there anything needed on user side?
> 
> My initial idea was about the need to restart the server though and I used Prometheus CR as an example of implementation which might caused misunderstanding.
> 
> Infinispan server itself doesn't need to be restarted to catch on updated credentials. If we had side container catching up on the secret changes we could trigger config-generator reload which would update all user.properties file on fly while server would be still up and running. But this is way out of scope of this PR

Yes, true, We would have side effect in the future, but better to implement as separate PR